lib/galaxy/datatypes/data.py

```diff
@@ -201,10 +202,10 @@ def display_peek(self, dataset ):
                 line = line.strip()
                 if not line:
                     continue
-                if isinstance(line, unicode):
+                if isinstance(line, text_type):
                     out.append( '<tr><td>%s</td></tr>' % escape( line ) )
                 else:
-                    out.append( '<tr><td>%s</td></tr>' % escape( unicode( line, 'utf-8' ) ) )
+                    out.append( '<tr><td>%s</td></tr>' % escape( text_type( line, 'utf-8' ) ) )
```

Comment:
This does not work in Python 3 because `text_type()` is just replaced with `str()`:
```
TypeError: decoding str is not supported
```